schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::cpwStateChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::CPWState cpwState){
+        qCDebug(dcSchneiderElectric()) << thing << "CPW state changed" << cpwState;
+        setCpwState(thing, cpwState);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::lastChargeStatusChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::LastChargeStatus lastChargeStatus){
+        qCDebug(dcSchneiderElectric()) << thing << "Last charge status changed" << lastChargeStatus;
+        setLastChargeStatus(thing, lastChargeStatus);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::stationEnergyChanged, this, [thing](quint32 stationEnergyInWattHours){
+        double stationEnergyInKilowattHours{(static_cast<double>(stationEnergyInWattHours)) / 1000.0};
+        qCDebug(dcSchneiderElectric()) << thing << "station energy changed" << stationEnergyInKilowattHours << "kWh";
+        thing->setStateValue("totalEnergyConsumed", stationEnergyInKilowattHours);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::stationPowerTotalChanged, this, [thing](float stationPowerTotalKilowatt){
+        double stationPowerTotalWatt{stationPowerTotalKilowatt * 1000};
+        qCDebug(dcSchneiderElectric()) << thing << "station total power changed" << stationPowerTotalWatt << "W";
+        thing->setStateValue("currentPower", stationPowerTotalWatt);
+        thing->setStateValue("charging", stationPowerTotalWatt > 20);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::errorStatusChanged, this, [thing, this](quint32 errorBits){
+        qCDebug(dcSchneiderElectric()) << thing << "Error state changed" << errorBits ;
+        setErrorMessage(thing, errorBits);
+    });
+
+    schneiderWallboxTcpConnection->connectDevice();
+
+    connect(schneiderWallbox, &SchneiderWallbox::phaseCountChanged, this, [thing, this](quint16 phaseCount){
+        qCDebug(dcSchneiderElectric()) << thing << "Phase count changed" << phaseCount ;
+        setPhaseCount(thing, phaseCount);
+    });
+
+    /* Probably not needed.
+    connect(schneiderWallbox, &SchneiderWallbox::currentPowerChanged, this, [thing, this](double currentPower){
+        qCDebug(dcSchneiderElectric()) << thing << "Current power changed" << currentPower ;
+        setCurrentPower(thing, currentPower);
+        thing->setStateValue("currentPower", currentPower);
+        thing->setStateValue("charging", currentPower > 20);
+    });
+    */
+}
+
+
+void IntegrationPluginSchneiderWallbox::postSetupThing(Thing *thing)
+{
+    qCDebug(dcSchneiderElectric()) << "Post setup" << thing->name();
+    if (thing->thingClassId() != schneiderEVlinkThingClassId) {
+        qCWarning(dcSchneiderElectric()) << "Thing class id not supported" << thing->thingClassId();
```

Comment:
same as setupThing(). can safely go away unless you plan to add additional thingClasses to the plugin.